### PR TITLE
Moved the `render text: ''` deprecation warning:

### PR DESCRIPTION
**Pull request by Edouard-chin**
- The deprecation warning was not picked up by ActionMailer as ActionController::Metal::Rendering is not included within ActionMailer
- Added the deprecation in the AbstractController::Rendering module instead

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Does `render :plain` work in action mailer?

**Comment by Edouard-chin** [comment]
TL;DR: `render plain: ''` on rails 5.1 will raise a `ActionView::MissingTemplate`.

On rails 5.1 `render plain: ''` will try to render the template based on the action name. 

There is a defensive check [here](https://github.com/rails/rails/blob/master/actionview/lib/action_view/renderer/template_renderer.rb#L41)  but `render plain: ''` wont enter inside. Because at the time `determine_template(options)` is called, the `options` hash contains a `template` key (the `template` key is set when [normalizing](https://github.com/rails/rails/blob/master/actionview/lib/action_view/rendering.rb#L149) the options)



```ruby
```

**Comment by rafaelfranca** [comment]
hmmm, so I think it is only implemented in 5.1. In that case we should deprecate `render :text` in 5.1 for mailer instead of removing it.

**Comment by Edouard-chin** [comment]
ok I'll have a stab at this. Wondering though how I can deprecate `render text: ''` only for mailers. It's so tightly coupled to ActionController rendering. Any hints?

**Comment by rafaelfranca** [comment]
We can define the same method on Action Mailer and call super.

**Comment by rafaelfranca** [comment]
This is confusing. So `render :text` in Action View will render a template with a text template for not? we should match the same behavior of Action View.

**Comment by Edouard-chin** [comment]
Yeah sorry for the confusion. So to be clear:

`render text: 'blabla'` doesn't **not** work on 5.1

`render plain: 'blabla'` **will work** on 5.1
